pkg/controller/core/workload_controller.go

```diff
@@ -127,6 +127,17 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	ctx = ctrl.LoggerInto(ctx, log)
 	log.V(2).Info("Reconciling Workload")
 
+	// if a pods ready timeout eviction is ongoing.
+	if evictionCond := apimeta.FindStatusCondition(wl.Status.Conditions, kueue.WorkloadEvicted); evictionCond != nil && evictionCond.Status == metav1.ConditionTrue &&
+		evictionCond.Reason == kueue.WorkloadEvictedByPodsReadyTimeout &&
```

Comment:
Maybe I said this before, but I think it shouldn't be conditional on the reason.
But we can standardize with preemption in a follow up, and based on the outcome of #510

Leaving this comment for future reference.